js/ui/components/networkAgent.js

```diff
@@ -140,6 +145,104 @@ var NetworkSecretDialog = new Lang.Class({
         this._updateOkButton();
     },
 
+    _addMeteredConnectionToggle: function(connection, layout, position) {
+        let updatesLayout = new Clutter.GridLayout({ orientation: Clutter.Orientation.VERTICAL });
+        let updatesTable = new St.Widget({ style_class: 'network-dialog-secret-table',
+                                           layout_manager: updatesLayout });
+        updatesLayout.hookup_style(updatesTable);
+
+        // Separator
+        let separator = new St.Widget({ style_class: 'popup-separator-menu-item',
+                                        y_expand: true,
+                                        y_align: Clutter.ActorAlign.CENTER });
+        updatesLayout.attach(separator, 0, 0, 3, 1);
+
+        // Icon
+        let iconFile = Gio.File.new_for_uri('resource:///org/gnome/shell/theme/endless-auto-updates-on-symbolic.svg');
+        let gicon = new Gio.FileIcon({ file: iconFile });
+        let icon = new St.Icon({ gicon: gicon,
+                                 icon_size: 24 });
+        updatesLayout.attach(icon, 0, 1, 1, 1);
+
+        // Title label
+        let title = new St.Label({ text: _("Is this connection paid or limited?"),
+                                   style_class: 'message-dialog-title',
+                                   x_expand: true,
+                                   x_align: Clutter.ActorAlign.START });
+        updatesLayout.attach(title, 1, 1, 1, 1);
+
+        // Subtitle label
+        let subtitle = new St.Label({ style_class: 'message-dialog-body',
+                                      text: _("Enable this option if this connection has usage restrictions, or is paid. This will disable automatic updates."),
```

Comment:
I really want to make sure we test this wording in market during the beta period.